Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -95,5 +128,310 @@ public static class CustomEntitySearch
+                                            currTextCharIndex++;
+                                        }
+                                    }
+
+                                }
+                            }
+                        }
+                        else
+                        {
+                            int initialOffsetIndex = currTextCharIndex - wordFound.ToString().Length;
+                            currWordCharIndex = 0;
+
+                            if (bestMismatchPreWhitespace > -1 && whitespaceOffset != initialOffsetIndex)
+                            {
+                                currTextCharIndex = whitespaceOffset;
+                            }
+                            wordFound.GetStringBuilder().Clear();
+                            currMismatch = 0;
+                            bestMismatchPreWhitespace = -1;
+                        }
+                        if (currWordCharIndex >= wordCharArray.Length && !textCharArray[currTextCharIndex - 1].IsDelineating())
+                        {
+                            while (currTextCharIndex < textCharArray.Length)
+                            {
+                                if (!textCharArray[currTextCharIndex].IsDelineating())
+                                {
+                                    wordFound.Write(text[currTextCharIndex]);
+                                    currTextCharIndex++;
+                                    currMismatch++;
+                                }
+                                else
+                                    break;
+                            }
+                        }
+                        if (currTextCharIndex >= textCharArray.Length || 
+                            (currWordCharIndex >= wordCharArray.Length && textCharArray[currTextCharIndex].IsDelineating()))
+                        {
+                            if (wordCharArray.Length - currWordCharIndex > 0)
+                                currMismatch += wordCharArray.Length - currWordCharIndex;
+                            int initialOffsetIndex = currTextCharIndex - wordFound.ToString().Length;
+                            if (currMismatch <= leniency)
+                            {
+                                if (bestMismatchPreWhitespace > -1 && bestMismatchPreWhitespace < currMismatch)
+                                {
+                                    entities.Add(
+                                    new Entity
+                                    {
+                                        Category = "customEntity",
+                                        Value = wordFound.GetStringBuilder().Remove(whitespaceOffset - initialOffsetIndex - 1, (int)bestMismatchPreWhitespace).ToString(),
+                                        Offset = initialOffsetIndex,
+                                        Confidence = leniency - bestMismatchPreWhitespace - (wordCharArray.Length - currWordCharIndex)
```

Comment:
> leniency - bestMismatchPreWhitespace - (wordCharArray.Length - currWordCharIndex) [](start = 52, length = 82)

what does this number convey?

if i get an exact match, it would return the leniency? Should this just be the mismatch value?